deltachat-ffi/deltachat.h

```diff
@@ -3603,6 +3652,31 @@ int dc_msg_get_videochat_type (const dc_msg_t* msg);
 #define DC_VIDEOCHATTYPE_JITSI       2
 
 
+/**
+ * Check if important parts of the message may be missing.
+ *
+ * dc_msg_is_mime_modified() checks
+ * if the original message _may_ contain important parts
+ * that are removed by some heuristics
+ * or if the message is just too long or too complex
+ * to get displayed properly by just using plain text.
+ * If so, the UI should offer a button as
+ * "Show full message" that shows the uncut message using dc_get_original_mime_html().
+ *
+ * Even if a "Show full message" button is recommended,
+ * the UI should display the text in the bubble
+ * using the normal dc_msg_get_text() function -
+ * which will still be fine in many cases.
+ *
+ * @memberof dc_msg_t
+ * @param msg The message object.
+ * @return 0=Message as displayed using dc_msg_get_text() is just fine;
+ *     1=The message may be cut badly
+ *     and a button to show the original should be offered
+ */
+int dc_msg_is_mime_modified (dc_msg_t* msg);
```

Comment:
This one I would name `dc_msg_has_html` or something like this, avoiding MIME. API users don't need to know anything about MIME, it's an implementation detail from the perspective of DC core API. UI developers don't care if email/MIME is used or matrix or XMPP or whatever to transport messages.

Reply:
first wording was is_maybe_badly_cut() or so :)  the current wording comes from an a/v with @dignifiedquire, but i agree, the look from the API-user and a simple has_html()/get_html() is maybe better, cc @dignifiedquire

Reply:
renamed the function to dc_msg_has_html()